### PR TITLE
React Revamp: styling react applications lesson

### PR DESCRIPTION
**Pull request by 01zulfi**
## Because

https://github.com/TheOdinProject/top-meta/issues/254

## This PR

Adds styling React applications lesson

## Issue

Closes https://github.com/TheOdinProject/top-meta/issues/254

## Pull Request Requirements

-   [ ] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [ ] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [ ] The `Because` section summarizes the reason for this PR
-   [ ] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [ ] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/the_react_ecosystem/styling_react_applications.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+Vanilla CSS is the simplest way to style. In the previous courses, you've learned a lot of CSS and all of those skills are applicable to React. There are a couple of things we'd like to highlight.
+
+1. CSS Modules
+
+    CSS Modules let you write CSS style declarations that are scoped locally.
+
+1. CSS Utility Frameworks
+
+    CSS frameworks are popular choice of styling React applications. They provide a set of pre-defined classes that you can directly use in your HTML, or JSX in our case. [Tailwind CSS](https://tailwindcss.com) is by far the most popular choice.
```
Comment (ManonLef) [inline react/the_react_ecosystem/styling_react_applications.md:21]:
```suggestion
Vanilla CSS is the simplest way to style. In the previous courses, you've learned a lot of CSS and all of those skills are applicable to React. There are a couple of things we'd like to highlight.

1. `CSS Modules` let you write CSS style declarations that are scoped locally.
1. `CSS Utility Frameworks` are a popular choice for styling React applications. They provide a set of pre-defined classes that you can directly use in your HTML, or JSX in our case. [Tailwind CSS](https://tailwindcss.com) is by far the most popular choice. 
```

The styling/layout of this seems "off" to me even though I know it adheres to the syleguide. I think the explanation just don't need their own subsection. I proposed something else but you may know of something better or want to leave it as is. 

do mind the typo that a fixed in the second list item.

File: react/the_react_ecosystem/styling_react_applications.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+1. CSS Utility Frameworks
+
+    CSS frameworks are popular choice of styling React applications. They provide a set of pre-defined classes that you can directly use in your HTML, or JSX in our case. [Tailwind CSS](https://tailwindcss.com) is by far the most popular choice. 
+
+
+### CSS in JS
+
+Why even write CSS in CSS when you can write it in JavaScript? Just kidding, of course! 
+
+CSS-in-JS is a paradigm for styling front-end projects. It allows you to take entire control of CSS with JavaScript and extends it with various features. Additionally, it also helps to apply styling in a logical fashion i.e. based on state. There are various CSS-in-JS solutions. One of the most popular ones in the React ecosystem is [styled-components](https://styled-components.com/).
```
Comment (ManonLef) [inline react/the_react_ecosystem/styling_react_applications.md:28]:
```suggestion
CSS-in-JS is a paradigm for styling front-end projects. It allows you to entirely take control of CSS with JavaScript and extends it with various features. Additionally, it also helps to apply styling in a logical fashion i.e. based on state. There are various CSS-in-JS solutions. One of the most popular ones in the React ecosystem is [styled-components](https://styled-components.com/).
```

nit, don't feel strongly about this. The sentence just didn't flow right for me reading it the first time.

File: react/the_react_ecosystem/styling_react_applications.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+### Component Libraries
+
+What if everything's already done for you? Styling, behavior, and accessibility are taken care for you in component libraries. As the name suggests, these libraries provide adaptable and reusable components that you can use directly in your project. These components include, but are not limited to, dropdowns, drawers, calendars, toggles, tabs, and all other components you can think of.
```
Comment (ManonLef) [inline react/the_react_ecosystem/styling_react_applications.md:32]:
```suggestion
What if everything's already done for you? Styling, behavior, and accessibility are taken care of for you in component libraries. As the name suggests, these libraries provide adaptable and reusable components that you can use directly in your project. These components include, but are not limited to, dropdowns, drawers, calendars, toggles, tabs, and all other components you can think of. 
```

File: react/the_react_ecosystem/styling_react_applications.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+### Component Libraries
+
+What if everything's already done for you? Styling, behavior, and accessibility are taken care for you in component libraries. As the name suggests, these libraries provide adaptable and reusable components that you can use directly in your project. These components include, but are not limited to, dropdowns, drawers, calendars, toggles, tabs, and all other components you can think of. 
+
+[Material UI](https://mui.com/), [Radix](https://www.radix-ui.com/), and [Chakra UI](https://chakra-ui.com/) are worth a mention when talking about component libraries.
+
+<div class="lesson-note lesson-note--tip" markdown="1" >
+
+For the purposes of learning and this course, we advise you to implement your component's styling from scratch i.e. use vanilla CSS or a CSS-in-JS option.
```
Comment (ManonLef) [inline react/the_react_ecosystem/styling_react_applications.md:38]:
```suggestion
For learning purposes throughout this course, we advise you to implement your component's styling from scratch i.e. use vanilla CSS or a CSS-in-JS option. 
```

`For the purposes of learning and this course` doesn't sound entirely right to me. Maybe you have another way to put it?

File: react/the_react_ecosystem/styling_react_applications.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+1. Skim through the [styled components documentation](https://styled-components.com/)
+
+</div>
+
+### Knowledge check
+
+This section contains questions for you to check your understanding of this lesson on your own. If you’re having trouble answering a question, click it and review the material it links to.
+
+- [How can you use CSS Modules in your React app?](https://www.makeuseof.com/react-components-css-modules-style/)
+- [What does CSS in JS mean?](#css-in-js)
```
Comment (ManonLef) [inline react/the_react_ecosystem/styling_react_applications.md:59]:
```suggestion
- [What does CSS-in-JS mean?](#css-in-js)
```
To stay consistent with the rest of the lesson

File: react/the_react_ecosystem/styling_react_applications.md (hunk 1 above)
Comment (01zulfi) [reply react/the_react_ecosystem/styling_react_applications.md:21]: lgtm

**Review by ManonLef — APPROVED** [review]
Looks great to me 👏🏻

**Pull request merged**